### PR TITLE
PR to make some path problems

### LINKED ISSUE #277: Consider Unix programming

### PR DESCRIPTION
**Pull request by danghai**
(Put an `X` inside the `[ ]` to denote check mark `[X]`.)

- [x] **If creating a new file :**
  - [x] added links to it in the README files ?
  - [x] included tests with it ?
  - [x] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

- [x] **if done some changes :**
  - [x] wrote short description in the PR explaining what the changes do ?
  - [x] Fixes #277 

- [ ] **other**

### CODE REVIEW

File: algorithms/unix/path/join_with_slash.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+http://algorithms.com/ + part --> http://algorithms.com/part
+http://algorithms.com + part --> http://algorithms/part
+"""
+import os
+
+def join_with_slash(base, suffix):
+    # Remove / trailing
+    base = base.rstrip('/')
+    # Remove / leading
+    suffix = suffix.lstrip().rstrip()
```
Comment (goswami-rahul) [inline algorithms/unix/path/join_with_slash.py]: should this be
```python
suffix = suffix.lstrip('/').rstrip()
```
?

File: algorithms/unix/path/split.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+def split(path):
+    parts = []
+    split_part = path.split('/')
+    # Remove the last element in split_part
+    split_part.pop(-1)
+    # Takt the origin path without the last part
+    parts.append("/".join(split_part))
+    # Take the last element of list
+    parts.append(path.split('/')[-1])
```
Comment (goswami-rahul) [inline algorithms/unix/path/split.py]: maybe using `path.rpartition('/')` can be easier ? [`str.rpartition`](https://docs.python.org/3/library/stdtypes.html#str.rpartition).

File: algorithms/unix/path/join_with_slash.py (hunk 1 above)
Comment (danghai) [reply algorithms/unix/path/join_with_slash.py]: Yeah. Thanks. I miss `/`

File: algorithms/unix/path/split.py (hunk 2 above)
Comment (danghai) [reply algorithms/unix/path/split.py]: This function `str.rpartition` is better

**Comment by danghai** [comment]
@goswami-rahul I have updated it.

**Pull request merged**